ui/components/component-library/popover/popover.tsx

```diff
@@ -10,124 +10,139 @@ import {
   Display,
   JustifyContent,
 } from '../../../helpers/constants/design-system';
-import Box from '../../ui/box/box';
-import { PopoverPosition, PopoverProps, PopoverRole } from '.';
 
-export const Popover = ({
-  children,
-  className = '',
-  position = PopoverPosition.Auto,
-  role = PopoverRole.Tooltip,
-  hasArrow = false,
-  matchWidth,
-  preventOverflow = false,
-  offset = [0, 8],
-  flip = false,
-  referenceHidden = true,
-  referenceElement,
-  isOpen,
-  title,
-  isPortal = false,
-  arrowProps,
-  onPressEscKey,
-  ...props
-}: PopoverProps) => {
-  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
-  const [arrowElement, setArrowElement] = useState<HTMLElement | null>(null);
+import { Box } from '..';
+import type { BoxProps, PolymorphicRef } from '../box';
 
-  // Define Popper options
-  const { styles, attributes } = usePopper(referenceElement, popperElement, {
-    placement: position,
-    modifiers: [
-      {
-        name: 'preventOverflow',
-        enabled: position === PopoverPosition.Auto ? true : preventOverflow,
-      },
-      {
-        name: 'flip',
-        enabled: position === PopoverPosition.Auto ? true : flip,
-      },
-      {
-        name: 'arrow',
-        enabled: hasArrow,
-        options: {
-          element: arrowElement,
+import {
+  PopoverProps,
+  PopoverComponent,
+  PopoverPosition,
+  PopoverRole,
+} from './popover.types';
+
+export const Popover: PopoverComponent = React.forwardRef(
+  <C extends React.ElementType = 'div'>(
+    {
+      children,
+      className = '',
+      position = PopoverPosition.Auto,
+      role = PopoverRole.Tooltip,
+      hasArrow = false,
+      matchWidth,
+      preventOverflow = false,
+      offset = [0, 8],
+      flip = false,
+      referenceHidden = true,
+      referenceElement,
+      isOpen,
+      title,
+      isPortal = false,
+      arrowProps,
+      onPressEscKey,
+      ...props
+    }: PopoverProps<C>,
+    ref?: PolymorphicRef<C>,
```

Comment:
Can I remove this line? It is not used anywhere in this file, and the code still functions correctly without it.

Reply:
> Can I remove this line? It is not used anywhere in this file, and the code still functions correctly without it.

No! You don't